ts/components/registration/stages/RestoreAccount.tsx

```diff
@@ -0,0 +1,321 @@
+                    iconColor="var(--text-primary-color)"
+                    style={{ margin: '-4px 0 0 8px' }}
+                  />
+                </Flex>
+                <SpacerSM />
+                <OnboardDescription>{window.i18n('onboardingRecoveryPassword')}</OnboardDescription>
+                <SpacerLG />
+                <SessionInput
+                  autoFocus={true}
+                  disabledOnBlur={true}
+                  type="password"
+                  placeholder={window.i18n('recoveryPasswordEnter')}
+                  value={recoveryPassword}
+                  onValueChanged={(seed: string) => {
+                    dispatch(setRecoveryPassword(seed));
+                    dispatch(
+                      setRecoveryPasswordError(
+                        !seed ? window.i18n('recoveryPasswordEnter') : undefined
+                      )
+                    );
+                  }}
+                  onEnterPressed={recoverAndFetchDisplayName}
+                  error={recoveryPasswordError}
+                  enableShowHide={true}
+                  inputDataTestId="recovery-phrase-input"
+                />
+                <SpacerLG />
+                <SessionButton
+                  buttonColor={SessionButtonColor.White}
+                  onClick={recoverAndFetchDisplayName}
+                  text={window.i18n('continue')}
+                  disabled={!(!!recoveryPassword && !recoveryPasswordError)}
+                  dataTestId="continue-button"
+                />
+              </>
+            ) : (
+              <Flex container={true} width="100%" flexDirection="column" alignItems="flex-start">
+                <OnboardHeading>{window.i18n('displayNameNew')}</OnboardHeading>
+                <SpacerSM />
+                <OnboardDescription>{window.i18n('displayNameErrorNew')}</OnboardDescription>
+                <SpacerLG />
+                <SessionInput
+                  autoFocus={true}
+                  disabledOnBlur={true}
+                  type="text"
+                  placeholder={window.i18n('enterDisplayName')}
+                  value={displayName}
+                  onValueChanged={(_name: string) => {
+                    const name = sanitizeDisplayNameOrToast(_name, setDisplayNameError, dispatch);
+                    dispatch(setDisplayName(name));
+                  }}
+                  onEnterPressed={recoverAndEnterDisplayName}
+                  error={displayNameError}
+                  inputDataTestId="display-name-input"
+                />
+                <SpacerLG />
+                <SessionButton
+                  buttonColor={SessionButtonColor.White}
+                  onClick={recoverAndEnterDisplayName}
+                  text={window.i18n('continue')}
+                  disabled={
+                    !(!!recoveryPassword && !recoveryPasswordError) ||
+                    !(!!displayName && !displayNameError)
+                  }
```

Comment:
this is pretty hard to understand.
it's not the equivalent of 
```suggestion
                  disabled={
                    !recoveryPassword || recoveryPasswordError || !displayName || displayNameError
                  }
```

Reply:
yep that's better